controller/settings.go

```diff
@@ -47,3 +47,35 @@ func (o OnConnectSettingsHandler) RouterConnected(r *network.Router) {
 		}).Info("no on connect settings to send")
 	}
 }
+
+type OnConnectCtrlAddressesUpdateHandler struct {
+	callback func() []string
+}
+
+func (o *OnConnectCtrlAddressesUpdateHandler) RouterDisconnected(r *network.Router) {
+	//do nothing, satisfy interface
+}
+
+func (o OnConnectCtrlAddressesUpdateHandler) RouterConnected(r *network.Router) {
```

Comment:
I'd probably do 
```
log := pfxlog.Logger().WithField("routerId", r.Id
```
at the top here and that way you don't have to do it at every point of use